src/components/Accordion/AccordionHeader.tsx

```diff
@@ -1,27 +1,29 @@
 /* (c) Copyright Frontify Ltd., all rights reserved. */
 
-import React, { FC, ReactElement, cloneElement, isValidElement } from 'react';
+import React, { ReactElement, cloneElement, isValidElement } from 'react';
 import { IconSize } from '@foundation/Icon/IconSize';
 import { IconProps } from '@foundation/Icon/IconProps';
 import { merge } from '@utilities/merge';
 import { AccordionHeaderIconSize, AccordionHeaderProps } from './types';
 import { AccordionHeaderIcon } from './AccordionHeaderIcon';
 
-export const AccordionHeader: FC<AccordionHeaderProps> = ({
+export const AccordionHeader = ({
     isOpen = false,
     decorator,
     disabled = false,
     children,
     size = AccordionHeaderIconSize.Medium,
     bold = true,
     type,
-    as: Heading = 'div',
-}) => {
+    as: Heading = 'span',
+}: AccordionHeaderProps) => {
     const icon = <AccordionHeaderIcon isOpen={isOpen} disabled={disabled} size={size} type={type} />;
     return (
-        <div data-test-id="accordion-header" className="tw-px-8 tw-py-6">
-            <header
+        <span data-test-id="accordion-header" className="tw-block tw-px-8 tw-py-6">
+            <span
                 data-test-id="fieldset-header"
+                role="navigation"
+                aria-label={`${children}`}
```

Comment:
Not correct as the type is `ReactNode`, you would have either `undefined`, either `false`/`true`, either `[Object object]`, ...
You need to compute it

Reply:
Same answer as the previous one.
The `children` prop is defined as `string` here https://github.com/Frontify/fondue/blob/beta/src/components/Accordion/types.ts#L38

Reply:
But it seems we're going to have this issue because of this:
https://github.com/Frontify/clarify/blob/master/application/client/Feature/Publishing/Controls/ImageGalleryControl/List/AccordionHeader/AccordionHeader.tsx#L21

Reply:
☝️ Actually not, because it is rendered a custom `AccordionHeader` component, not the Fondue one. But there will be the invalid HTML tree `button > div`.

Reply:
Anyway, I created a state ref for the former `Heading` (which was coming from the deprecated `as` property and will be removed from the `AccordionHeaderProps` in another PR) and added as ref to the new `span` tag which will render `children` and I'm getting the `innerText` of that element.